sw/device/silicon_creator/rom/bootstrap.c

```diff
@@ -168,14 +169,11 @@ static rom_error_t bootstrap_page_program(uint32_t addr, size_t byte_count,
   }
 
   // Round up to next flash word and fill missing bytes with `0xff`.
-  size_t flash_word_misalignment = byte_count & kFlashWordMask;
-  if (flash_word_misalignment > 0) {
-    size_t padding_byte_count =
-        FLASH_CTRL_PARAM_BYTES_PER_WORD - flash_word_misalignment;
-    for (size_t i = 0; i < padding_byte_count; ++i) {
-      data[byte_count++] = 0xff;
-    }
-  }
```

Comment:
I like how your version uses `memset` instead of a for loop. WDYT about not introducing the `compute_padding()` but rewriting this part as

```
size_t flash_word_misalignment = byte_count & kFlashWordMask;
size_t padding_byte_count = -misalignment & kFlashWordMask;
memset(&data[byte_count], 0xff, padding_byte_count);
byte_count += padding_byte_count;
```

Since we know that `kFlashWordMask` is a power of two minus 1 (lower bits are all ones), this simplifies to

```
size_t padding_byte_count = -byte_count & kFlashWordMask;
memset(&data[byte_count], 0xff, padding_byte_count);
byte_count += padding_byte_count;
```

Reply:
as we discussed offline, I'm happy with adding `compute_padding` as long as it expect a power of two minus 1 mask and doesn't use the modulo operator (not an issue if block_size is power of two but pulls in a large division function if not and it might be hard to enforce.).

Reply:
I may be able to do one better (subjectively) with a macro. This would enable us to statically assert that the block size is a power of two. WDYT?

Reply:
Nevermind, it's quite difficult to put static assertions into a macro that evaluates as an expression without any C++ magic. I think that your suggestion of changing the parameter to a mask is a good middle ground.